python/ray/air/integrations/wandb.py

```diff
@@ -516,18 +519,44 @@ def log_trial_start(self, trial: "Trial"):
         )
         wandb_init_kwargs.update(self.kwargs)
 
-        self._trial_queues[trial] = Queue()
-        self._trial_processes[trial] = self._logger_process_cls(
+        self._start_logging_actor(trial, exclude_results, **wandb_init_kwargs)
+
+    def _start_logging_actor(
+        self, trial: "Trial", exclude_results: List[str], **wandb_init_kwargs
+    ):
+        if not self._remote_logger_class:
+            self._remote_logger_class = ray.remote(
+                num_cpus=0, **_force_on_current_node()
+            )(self._logger_actor_cls)
+
+        self._trial_queues[trial] = Queue(
+            actor_options={"num_cpus": 0, **_force_on_current_node()}
+        )
+        self._trial_logging_actors[trial] = self._remote_logger_class.remote(
             logdir=trial.logdir,
             queue=self._trial_queues[trial],
             exclude=exclude_results,
             to_config=self._config_results,
             **wandb_init_kwargs,
         )
-        self._trial_processes[trial].start()
+        self._trial_logging_futures[trial] = self._trial_logging_actors[
+            trial
+        ].run.remote()
+
+    def _stop_logging_actor(self, trial: "Trial", timeout: int = 10):
+        self._trial_queues[trial].put((_QueueItem.END, None))
+
+        try:
+            ray.get(self._trial_logging_futures[trial], timeout=timeout)
+        except TimeoutError:
+            ray.kill(self._trial_logging_actors[trial])
```

Comment:
should we keep this method isolated to just the ray.get/ray.kill logic? That way, in the test we can just override the ray related logic only, and don't have to override the `.put` or the `del` logic.

Reply:
I think it's ok to keep this as is as the test also uses threads and calls some thread-specific methods (join)